tools/js-sdk-release-tools/src/autoGenerateInPipeline.ts

```diff
@@ -51,13 +54,20 @@ async function automationGenerateInPipeline(
                     swaggerRepoUrl: repoHttpsUrl,
                     downloadUrlPrefix: downloadUrlPrefix,
                     skipGeneration: skipGeneration,
-                    runningEnvironment: runningEnvironment
+                    runningEnvironment: runningEnvironment,
+                    apiVersion: apiVersion,
+                    sdkReleaseType: sdkReleaseType,
                 });
                 break;
             case SDKType.RestLevelClient:
+                const swaggerRepo = path.isAbsolute(specFolder) ? specFolder : path.join(String(shell.pwd()), specFolder)
+                if (typespecProject && !skipGeneration && sdkGenerationType !== "command") {                    
+                    const tspDefDir = path.join(swaggerRepo, typespecProject);
+                    updateApiVersionInTspConfig(tspDefDir, apiVersion);                       
```

Comment:
I think you can move this function before the `switch` block and and only update tspconfig.yaml when the sdk is going to generate from tsp. 
We should seperate by generate from tsp or swagger. 
And in next pr we will support swagger.